src/calculatePlayerMoves.ts

```diff
@@ -0,0 +1,243 @@
+  if (square.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+
+  // When there is a wall between player and targeted square (space closest to player)
+  let coordinatesWallInBetween: Coordinates = {
+    y: coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 4, // division by 4, because coordinate difference between player and targeted square is 4 and we select the square closest to the player
+    x: coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 4
+  };
+  let wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  // When there no other player between player and targeted square
+  const coordinatesPlayerInBetween: Coordinates = {
+    y: coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2, // division by 2, because coordinate difference between player and targeted square is 4 and we select the middle square (x / 4 * 2 = x / 2)
+    x: coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2
+  };
+  const playerInBetween = getSquareByCoordinates(coordinatesPlayerInBetween, board);
+  if (playerInBetween.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  if (!playerInBetween.playerId) return false;
+
+  // When there is a wall between player and targeted square (space closest to targeted square)
+  coordinatesWallInBetween = {
+    y: coordinatesPlayer.y + ((coordinatesSquare.y - coordinatesPlayer.y) / 4) * 3, // division by 4 and multiplication by 3, because coordinate difference between player and targeted square is 4 and we select the square between the middle square and targeted square
+    x: coordinatesPlayer.x + ((coordinatesSquare.x - coordinatesPlayer.x) / 4) * 3
+  };
+  wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  return true;
+};
+
+// Checks if a player can jump to a square diagonally.
+// It starts by checking horizontally if argument 'startFromHorizontal' is set to true.
+// There has to be other player blocking the way in selected direction, wall behind it (to block linear jump)
+// and no walls to prevent jumping diagonally.
+const checkDiagonalFromOneSide = (
+  coordinatesSquare: Coordinates,
+  coordinatesPlayer: Coordinates,
+  board: Board,
+  startFromHorizontal: boolean
+): boolean => {
+  const square = getSquareByCoordinates(coordinatesSquare, board);
+  if (square.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  // When targeted square is occupied by player
+  if (square.playerId) return false;
+
+  // When wall to the side of player's origin square is not walkable
+  let coordinatesWallInBetween: Coordinates = {

```

Comment:
The more I read the code, the more confusing this name gets. Name each coordinate differently. For example this could be `coordinatesDirectlyBlockingWall`